extensions/resource-deployment/src/ui/notebookWizard/notebookWizardPage.ts

```diff
@@ -96,7 +96,12 @@ export class NotebookWizardPage extends ResourceTypePage {
 			this.wizard.wizardObject.generateScriptButton.hidden = true;
 		}
 
+		// by default the generate button is disabled and will wait for the page validation to be enabled.
```

Comment:
All these matter only on last page because on all other pages generateScript button is hidden so you can move this logic within the pageInfo.isLastPage check above.